src/Equations/Fluid/ParallelHeatLossTerm.cpp

```diff
@@ -0,0 +1,245 @@
+*/
+real_t ParallelHeatLossTerm::StepFunction(len_t ir){
+    if ((int_t)ir > this->ir_LCFS) { 
+        return 1.; 
+    } else { 
+        return 0.; 
+    }
+}
+
+
+/** 
+* SETWEIGHTS: Needed for DiagonalTerm, defined similar to AvalancheGrowthTerm 
+*/
+void ParallelHeatLossTerm::SetWeights() {
+    FindRadiusOfLCFS(); // Find ir_LCFS
+    const real_t *jtot = this->unknowns->GetUnknownData(id_jtot);
+	
+    real_t *T_cold = unknowns->GetUnknownData(id_T_cold); 
+    real_t *N_i = unknowns->GetUnknownData(id_N_i); 
+    real_t *W_i = unknowns->GetUnknownData(id_W_i); 
+
+
+
+    for (len_t ir = 0; ir < nr; ir++) {
+	   	real_t T_i = 2. / 3. * W_i[ir] / N_i[ir]; 
+        real_t T_e = T_cold[ir] * Constants::ec; 
+        real_t mu0Ip = Constants::mu0 * TotalPlasmaCurrentFromJTot::EvaluateIpInsideR(ir,this->grid->GetRadialGrid(),jtot);
+    	real_t qR0 = this->grid->GetRadialGrid()->SafetyFactorNormalized(ir,mu0Ip);
+
+        this->weights[ir] =  StepFunction(ir) * 2. / 3. * kappa  * sqrt((T_e + gamma * T_i) / m_i) / (M_PI * qR0); 
+    }
+}
+
+
+/**
+ * SETDIFFWEIGHTS: Needed for DiagonalComplexTerm
+ */
+void ParallelHeatLossTerm::SetDiffWeights(len_t derivId, len_t nMultiples) {
+    // Retrieve necessary arrays from the unknowns handler
+    real_t *T_cold = unknowns->GetUnknownData(id_T_cold);
+    real_t *W_i = unknowns->GetUnknownData(id_W_i);
+    real_t *N_i = unknowns->GetUnknownData(id_N_i);
+    const real_t *jtot = unknowns->GetUnknownData(id_jtot);
+
+    // Calculate the electron temperature at the radial index 'nr'
+    real_t T_e = T_cold[nr] * Constants::ec;
+
+    auto calculateMu0Ip = [&](len_t ir) {
+        return Constants::mu0 * TotalPlasmaCurrentFromJTot::EvaluateIpInsideR(ir, grid->GetRadialGrid(), jtot);
+    };
+
+    auto calculateQ = [&](len_t ir) {
+        return grid->GetRadialGrid()->SafetyFactorNormalized(ir, calculateMu0Ip(ir));
+    };
+
+    auto calculateWeight = [&](len_t ir, real_t qR0, real_t coeff) {
+
+
+        return - StepFunction(ir) * coeff / (m_i * N_i[ir] * N_i[ir] * sqrt((T_e + gamma * 2. / 3. * W_i[ir] / N_i[ir]) / m_i)) / (2 * M_PI * qR0);
+    };
+
+    real_t coeff = (derivId == id_T_cold) ? (2. / 3.) : (4. / 9. * kappa * gamma);
+    
+    for (len_t i_ion = 0; i_ion <= nMultiples; i_ion++) {
+        for (len_t ir = 0; ir < nr; ir++) {
+            real_t qR0 = calculateQ(ir);
+            if (derivId == id_N_i) {
+                this->diffWeights[i_ion * nr + ir] = calculateWeight(ir, qR0, coeff * W_i[ir]);
+            } else if (derivId == id_T_cold) {
+                this->diffWeights[ir] = - calculateWeight(ir, qR0, coeff * W_i[ir]);
+            } else if (derivId == id_W_i){
+                this->diffWeights[i_ion * nr + ir] = - calculateWeight(ir, qR0, coeff);
+            }
+        }
+    }
+}
+
+
+
+/**
+* Return the weights on the radial grid
+*/
+const real_t* ParallelHeatLossTerm::GetParallelHeatLossWeights(){
+    return this->HeatLoss;
```

Comment:
The quantity ``HeatLoss`` always seems to be uninitialized and unused? Shouldn't this method return ``this->weights`` instead?